data/summary/types/summary.go

```diff
@@ -177,14 +172,22 @@ func NewConfig() Config {
 	}
 }
 
-func (s *Summary[T, A]) SetOutdated() {
+func (s *Summary[T, A]) SetOutdated(reason string) {
+	set := mapset.NewSet[string](reason)
+	if len(s.Dates.OutdatedReason) > 0 {
+		set.Append(s.Dates.LastUpdatedReason...)
+	}
+
+	s.Dates.LastUpdatedReason = nil
+	s.Dates.OutdatedReason = set.ToSlice()
```

Comment:
Is this worth bringing in an external dependency? The golang-set module looks like a nice module, I'll grant. But this could just be:

```go
func (s *Summary[T, A]) SetOutdated(reason string) {
	set := map[string]struct{}{}
	uniq := []string{}
	for _, r := range s.Dates.OutdatedReason {
		if _, found := set[r]; !found {
			set[r] = struct{}{}
			uniq = append(uniq, r)
		}
	}

	s.Dates.LastUpdatedReason = nil
	s.Dates.OutdatedReason = uniq
	s.Dates.OutdatedSince = pointer.FromAny(time.Now().Add(setOutdatedBuffer).UTC().Truncate(time.Millisecond))
	s.Dates.HasOutdatedSince = true
}
```
You could fix up my poor names a bit, and even extract it to a function if you want, but the point is, for such a simple set operation, I don't think bringing in an external dependency is worth it.

Reply:
I'm aware that I can use a map to track unique elements, but I find the version with the external dependency much easier to read and understand.

Reply:
This is such a common pattern we keep having all over the place, I for one welcome this external dependency, we can remove more than just the pattern here, but in a few other places in summary code next pass.